src/hooks/contenful/contentful.ts

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from "react"
+import {
+  ContentfulAsset,
+  ContentfulResponse,
+  Locales,
+  LocalizedField,
+  LocalizedFieldType,
+  SysAssetLink,
+} from "./contentful.types"
+
+const getAssetUrl = (
+  assets: Record<string, ContentfulAsset>,
+  locale: Locales,
+  assetLink?: SysAssetLink
+): string => {
+  if (!assetLink) return ""
+  const asset = assets[assetLink.sys.id]
+  return asset?.fields.file[locale]?.url
+    ? `https:${asset.fields.file[locale].url}`
+    : ""
+}
+
+const useGetContentfulEntry = <
+  T extends Record<string, LocalizedField<LocalizedFieldType>>,
+>(
+  id: string,
+  environment: string,
+  token: string,
+  space: string
+): {
+  fields: T | null
+  assets: {}
+  isLoading: boolean
+  error: string | null
+} => {
+  const [fields, setFields] = useState<T | null>(null)
+  const [assets, setAssets] = useState<Record<string, ContentfulAsset>>({})
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          `https://cdn.contentful.com/spaces/${space}/environments/${environment}/entries/?` +
+            new URLSearchParams({
+              "sys.id": id,
+              content_type: "banner",
```

Comment:
can I use `useGetContentfulEntry` to get other kind of entries, right?
Maybe we can pass it throw the props or, change the same of the hook to `useGetContentfulBannerEntry`. Or even remove the `content_type`

Reply:
Changed it so any content type can be used.